suitcase/fields.py

```diff
@@ -656,6 +725,51 @@ def setval(self, value):
         return self._get_parent_field().setval(value)
 
 
+class FieldArray(BaseField):
+    """Field which contains a list of some other field.
+
+    In some protocols, there exist repeated substructures which are present in a
+    variable number. The variable nature of these fields make a DispatchField
+    and DispatchTarget combination unsuitable; instead a FieldArray may be
+    used to pack/unpack these fields to/from a list.
+

```

Comment:
I like your example, but I'm having a hard time getting structures to embed properly (regardless of my pull request). Can you let me know if I'm doing something inherently incorrect in this test?

```{.py}
from suitcase.fields import LengthField, UBInt16, Payload
from suitcase.structure import Structure


class PascalString16(Structure):
    length = LengthField(UBInt16())
    value = Payload(length)


class FamousQuoteRecord(Structure):
    length = LengthField(UBInt16())
    value = Payload(length)

    author = PascalString16()


m = FamousQuoteRecord()
m.value = b"John Doe"
m.author.value = b"Jane Doe"

print(m.author.pack())  # Prints b"\x00\x08Jane Doe"
print(m.pack())  # Should print b"\x00\x08John Doe\x00\x08Jane Doe", but only prints b"\x00\x08John Doe"
```